src/devices/Bmp180/Bmp180.cs

```diff
@@ -0,0 +1,303 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+//Ported from https://github.com/adafruit/Adafruit_Python_BMP/blob/master/Adafruit_BMP/BMP085.py  
+//Formulas and code examples can also be found in the datasheet https://cdn-shop.adafruit.com/datasheets/BST-BMP180-DS000-09.pdf
+
+using System;
+using System.Device.I2c;
+using System.Buffers.Binary;
+using System.Threading;
+
+namespace Iot.Device.Bmp180
+{
+    public class Bmp180:IDisposable
+    {
+        private I2cDevice _i2cDevice;
+        private readonly CommunicationProtocol _communicationProtocol;
+        private bool _initialized = false;
+        private readonly CalibrationData _calibrationData;
+        private Sampling _mode = Sampling.Standard;
+
+        private enum CommunicationProtocol
+        {
+            I2c
+        }
+
+        public Bmp180(I2cDevice i2cDevice)
+        {
+            _i2cDevice = i2cDevice;            
+            _calibrationData = new CalibrationData();
+            _communicationProtocol = CommunicationProtocol.I2c;
+        }
+
+        private void Begin()
+        {            
+            //Read the coefficients table
+            _calibrationData.ReadFromDevice(this);
+            _initialized = true;
+        }
+
+        /// <summary>
+        /// Sets sampling to the given value
+        /// </summary>
+        /// <param name="mode">Sapling Mode</param>
+        public void SetSampling(Sampling mode)
+        {
+            this._mode = mode;
+        }
+
+        /// <summary>
+        ///  Reads the temperature from the sensor
+        /// </summary>
+        /// <returns>
+        ///  Temperature in degrees celsius
+        /// </returns>
+        public double ReadTemperature()
+        {
+            // Make sure the I2C device is initialized
+            if (!_initialized)
+            {
+                Begin();
+            }
+            //Gets the compensated temperature in degrees celsius
+            short UT = ReadRawTemperature();
+
+            // Calculations below are taken straight from section 3.5 of the datasheet.
+            int X1 = (UT - _calibrationData.AC6) * _calibrationData.AC5 / (1 << 15);
+            int X2 = _calibrationData.MC * (1 << 11) / (X1 + _calibrationData.MD);
+            int B5 = (X1 + X2);
+            int T = (B5 + 8) / (1 << 4);
+            
+            return (double)T/10;
+        }
+
+        /// <summary>
+        ///  Reads the pressure from the sensor
+        /// </summary>
+        /// <returns>
+        ///  Atmospheric pressure in Pa
+        /// </returns>
+        public int ReadPressure()

```

Comment:
Should this return a double instead?

Reply:
you're right, the return should be double

Reply:
I think this got missed on the last iteration